webui/module/Restore/src/Restore/Form/RestoreForm.php

```diff
@@ -599,137 +435,95 @@ public function __construct($restore_params = null, $clients = null, $filesets =
-                        $level = "Differential";
-                        break;
-                    case 'F':
-                        $level = "Full";
-                        break;
-                }
-                $selectData[$backup['jobid']] = "(" . $backup['jobid']  . ") " . $backup['starttime'] . " - " . $backup['name'] . " - " . $level;
+        if (!empty($source)) {
+            foreach ($source as $i) {
+                $selectData[$i["name"]] = $i["name"];
             }
         }
+        ksort($selectData);
         return $selectData;
     }
 
     /**
      *
      */
-    private function getJobIds()
-    {
-        return $this->jobids;
-    }
-
-    /**
-     *
-     */
-    private function getClientList()
+    private function getJobList()
     {
         $selectData = array();
-        if (!empty($this->clients)) {
-            foreach ($this->clients as $client) {
-                $selectData[$client['name']] = $client['name'];
+        if (!empty($this->jobs)) {
+            foreach ($this->jobs as $job) {
+                $selectData[$job['jobid']] = $job['jobid'] . " - " . $job['name'];
             }
         }
-        ksort($selectData);
         return $selectData;
     }
 
     /**
      *
      */
-    private function getFilesetList()
+    private function getBackupList()
     {
         $selectData = array();
-        if (!empty($this->filesets)) {
-            foreach ($this->filesets as $fileset) {
-                $selectData[$fileset['name']] = $fileset['name'];
+        foreach ($this->backups as $backup) {
+            switch ($backup['level']) {
+                case 'I':
+                    $level = "Incremental";
+                    break;
+                case 'D':
+                    $level = "Differential";
+                    break;
+                case 'F':
+                    $level = "Full";
+                    break;
             }
+            $selectData[$backup['jobid']] = "(" . $backup['jobid']  . ") " . $backup['starttime'] . " - " . $backup['name'] . " - " . $level;
         }
         return $selectData;
     }
 
     /**
      *
      */
-    private function getRestoreJobList()
+    private function getJobIds()
     {
-        $selectData = array();
-        if (!empty($this->restorejobresources)) {
-            foreach ($this->restorejobresources as $restorejob) {
-                $selectData[$restorejob['job']] = $restorejob['job'];
-            }
-        }
-        return $selectData;
+        return $this->jobids;
     }
 
     /**
      *
      */
-    private function getRestoreJobWhereDirectives()
+    private function getRestoreJobList()
     {
         $selectData = array();
-        if (!empty($this->restorejobresources)) {
-            foreach ($this->restorejobresources as $restorejob) {
-                $selectData[$restorejob['where']] = $restorejob['where'];
-            }
+        foreach ($this->restorejobresources as $key => $value) {
```

Comment:
For the same reason we also need an empty check here i believe.

Reply:
done